crates/proof-of-sql/src/sql/proof/provable_query_result.rs

```diff
@@ -55,6 +55,13 @@ impl ProvableQueryResult {
         }
     }
 
+    /// Form intermediate query result from a table
+    #[must_use]
+    pub fn new_from_table<S: Scalar>(table: &Table<S>) -> Self {
+        let columns = table.columns().copied().collect::<Vec<_>>();
```

Comment:
Not sure if this would work, but could you just derive `Clone` for `Table`, and then pass `table` by value instead of reference? Then, you can `.clone` upstream only if needed.

Then, you could have `Table` also implement `IntoIterator` and I think do something like `let columns = Vec::from_iter(table)` instead.

Reply:
@JayWhite2357 We can though that won't be intuitive since naturally `Item` here should be `(Identifier, Column)` as opposed to `Column`. If you want that instead I can do it but I need to document this.